operator/pkg/model/ingestion/gateway.go

```diff
@@ -480,11 +480,18 @@ func toHTTPRequestRedirectFilter(redirect *gatewayv1.HTTPRequestRedirectFilter)
 			pathModifier.Prefix = *redirect.Path.ReplacePrefixMatch
 		}
 	}
+	// If the redirectPort is not specified, use the listener's port
+	var redirectPort *int32
+	if redirect.Port == nil {
+		redirectPort = model.AddressOf(int32(listener.Port))
+	} else {
+		redirectPort = (*int32)(redirect.Port)
+	}
```

Comment:
The `root cause`  of the issue is here .

https://github.com/cilium/cilium/blob/7c7ae03aa95620c4b022dd1defcf2b7d424f296a/operator/pkg/model/translation/cec_translator.go#L287

If a gateway has multiple listeners, then there will be multiple `m.HTT`P here, while the current code always uses the first `m.HTTP`.

The reason I didn't modify `L287` is because I think my current approach may be more elegant.

The core idea of this PR is to leverage the following code:

https://github.com/cilium/cilium/blob/7c7ae03aa95620c4b022dd1defcf2b7d424f296a/operator/pkg/model/translation/envoy_virtual_host.go#L410-L422

Reply:
Awesome finding :+1:

Reply:
Will take a closer look later today or tmr

Reply:
Reference  https://github.com/kubernetes-sigs/gateway-api/blob/35fe25d1384a41c9b89dd5af7ae3214c431f008c/apis/v1/httproute_types.go#L1040-L1041